pandas/core/arrays/integer.py

```diff
@@ -546,9 +546,9 @@ def _reduce(self, name, skipna=True, **kwargs):
         if name in ['any', 'all']:
             pass
 
-        # if we have a numeric op, provide coercion back to an integer
-        # type if possible
-        elif notna(result):
+        # if we have a preservable numeric op,
+        # provide coercion back to an integer type if possible
+        elif name in ['sum', 'min', 'max'] and notna(result):
```

Comment:
We should maybe add a test for this specific aspect? (int vs foat return depending on the op)

Reply:
> we already have them

But not tests that assert the correct type of the result? Because you changed the behaviour here for certain methods, but didn't need to update any tests, so that seems to indicate this aspect is not covered?

Reply:
https://github.com/pandas-dev/pandas/pull/22762/files#diff-3523d268da9f1c7d4c82d4d27241ed07L590

these were already there, just expanded

Reply:
I added a test that I tried to ask for. Can you have a look at it? 

There is actually some more "complexity" here in the return value. If you access a single value from an IntegerArray, that gives you a np.int64 scalar (or other dtype). Now, the reductions here return a Python int.
I think we should try to be consistent here? (or for follow-up)